src/main/java/com/synopsys/integration/alert/channel/jira/JiraIssueHandler.java

```diff
@@ -192,12 +169,64 @@ public JiraMessageResult createOrUpdateIssues(JiraIssueConfig jiraIssueConfig, M
         return combinedItems;
     }
 
-    private Optional<IssueComponent> retrieveExistingIssue(String jiraProjectKey, String provider, LinkableItem topic, Optional<LinkableItem> subTopic, ComponentItem componentItem, String alertIssueUniqueId) throws IntegrationException {
-        final Optional<IssueSearchResponseModel> optionalIssueSearchResponse = jiraIssuePropertyHelper.findIssues(jiraProjectKey, provider, topic, subTopic.orElse(null), componentItem, alertIssueUniqueId);
-        return optionalIssueSearchResponse
+    private Set<IssueComponent> updateExistingIssues(List<IssueComponent> existingIssues, ItemOperation operation, JiraIssueConfig jiraIssueConfig, ComponentItem arbitraryItem, String providerName, Collection<ComponentItem> combinedItems)
+        throws IntegrationException {
+        Set<IssueComponent> updatedIssues = new HashSet<>();
+        List<IssueComponent> issuesToUpdate = filterUpdatableIssues(existingIssues, operation);
+        for (IssueComponent issue : issuesToUpdate) {
+            if (jiraIssueConfig.getCommentOnIssues()) {
+                String operationComment = createOperationComment(operation, arbitraryItem.getCategory(), providerName, combinedItems);
+                addComment(issue.getKey(), operationComment);
+                updatedIssues.add(issue);
+            }
+
+            boolean didUpdateIssue = jiraTransitionHelper.transitionIssueIfNecessary(issue.getKey(), jiraIssueConfig, operation);
+            if (didUpdateIssue) {
+                updatedIssues.add(issue);
+            }
+        }
+        return updatedIssues;
+    }
+
+    private IssueResponseModel createIssue(IssueRequestModelFieldsBuilder initialFieldsBuilder, JiraIssueConfig jiraIssueConfig, String providerName, LinkableItem topic, Optional<LinkableItem> subTopic, ComponentItem arbitraryItem,
+        String trackingKey) throws IntegrationException {
+        initialFieldsBuilder.setProject(jiraIssueConfig.getProjectComponent().getId());
+        initialFieldsBuilder.setIssueType(jiraIssueConfig.getIssueType());
+        String issueCreator = jiraIssueConfig.getIssueCreator();
+        IssueResponseModel issue = null;
+        try {
+            issue = issueService.createIssue(new IssueCreationRequestModel(issueCreator, jiraIssueConfig.getIssueType(), jiraIssueConfig.getProjectComponent().getName(), initialFieldsBuilder, List.of()));
+            logger.debug("Created new Jira Cloud issue: {}", issue.getKey());
+        } catch (IntegrationRestException e) {
+            handleIssueCreationRestException(e, issueCreator);
+        }
+        if (issue == null || StringUtils.isBlank(issue.getKey())) {
+            throw new AlertException("There was an problem when creating this issue.");
```

Comment:
Is there a case where an issue can be created without a key? If not, we probably just want to throw this exception in the catch statement above.

Reply:
This code was moved from a different method.